components-sdk/src/drag/DragEvents.tsx

```diff
@@ -0,0 +1,66 @@
+import { FC, ReactNode, useEffect } from 'react';
+import { StateManager } from '../polyfills/StateManager';
+import { DragContextType } from './types';
+
+import { useDragContext } from './DragContext';
+import { handleDragDrop } from './handleDragDrop';
+import { handleDragOver } from './handleDragOver';
+
+const handleDragStart = (e: DragEvent) => {
+    // Some items (like images) have a default dragstart event that we need to prevent
+    e.preventDefault();
+};
+
+const handleDragEnd = (
+    e: DragEvent,
+    {
+        setVisible,
+    }: {
+        setVisible: DragContextType['setVisible'];
+    }
+) => {
+    e.preventDefault();
+    setVisible(null);
+};
+
+export const DragEvents: FC<{
+    children: ReactNode;
+    stateManager: StateManager;
+}> = ({ children, stateManager }) => {
+    const { refs, visible, setVisible, keyToDelete } = useDragContext();
+
+    useEffect(() => {
+        const onDragOver = (e: DragEvent) => handleDragOver(e, {
+            refs,
+            visible,
+            setVisible,
+        });
+        const onDragStart = (e: DragEvent) => handleDragStart(e);
+        const onDragEnd = (e: DragEvent) => handleDragEnd(e, { setVisible });
+        const onDragDrop = (e: DragEvent) => handleDragDrop(e, {
+            visible,
+            setVisible,
+            stateManager,
+            keyToDelete,
+        });
+        const onMouseUp = () => setVisible(null);
+
+        window.addEventListener('dragstart', onDragStart);
+        window.addEventListener('dragover', onDragOver);
+        window.addEventListener('dragend', onDragEnd);
+        window.addEventListener('drop', onDragDrop);
+        window.addEventListener('mouseup', onMouseUp);
+
+        return () => {
+            window.removeEventListener('dragstart', onDragStart);
+            window.removeEventListener('dragover', onDragOver);
+            window.removeEventListener('dragend', onDragEnd);
+            window.removeEventListener('drop', onDragDrop);
+            window.removeEventListener('mouseup', onMouseUp);
+        };
```

Comment:
I still have to test this, but would it be possible to also have this work on mobile by registering to `touchstart`, `touchmove` and `touchend`?

Reply:
It wouldn't be that simple. Event `dragstart` is managed by [HTML Drag and Drop API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API) and gives you out of the box features like drop preview
![drop preview](https://github.com/user-attachments/assets/cb89da61-f79e-4b7c-8226-68e9232e4d46)

There are more serious problems for mobile right now tbh. (#24)